lib/mayaUsd/render/vp2RenderDelegate/render_delegate.cpp

```diff
@@ -361,7 +354,25 @@ namespace
     MSamplerStateCache sSamplerStates;  //!< Sampler state cache
     tbb::spin_rw_mutex sSamplerRWMutex; //!< Synchronization used to protect concurrent read from serial writes
 
+    MShaderCache sShaderCache;                      //!< Global shader cache to minimize the number of unique shaders.
     const HdVP2BBoxGeom* sSharedBBoxGeom = nullptr; //!< Shared geometry for all Rprims to display bounding box
+    std::once_flag sInitializeVP2ResourcesCalled;   //!< Whether or not InitializeVP2Resources() has been called.
+
+    void InitializeVP2Resources()
+    {
+        // Shader fragments can only be registered until VP2 has been initialized, thus the function
```

Comment:
I think this should maybe be: "Shader fragments can only be registered **after** VP2 has been initialized"? As is, it sounds like you have to register fragments before, which is the opposite.